bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.resol.handler;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.TimeZone;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.resol.internal.ResolBindingConstants;
+import org.openhab.binding.resol.internal.ResolStateDescriptionOptionProvider;
+import org.openhab.binding.resol.internal.providers.ResolChannelTypeProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.resol.vbus.Packet;
+import de.resol.vbus.Specification;
+import de.resol.vbus.Specification.PacketFieldSpec;
+import de.resol.vbus.Specification.PacketFieldValue;
+import de.resol.vbus.SpecificationFile;
+import de.resol.vbus.SpecificationFile.Enum;
+import de.resol.vbus.SpecificationFile.EnumVariant;
+import de.resol.vbus.SpecificationFile.Language;
+
+/**
+ * The {@link ResolThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class ResolThingHandler extends ResolBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ResolThingHandler.class);
+
+    @Nullable
+    ResolBridgeHandler bridgeHandler;
+    private ResolStateDescriptionOptionProvider stateDescriptionProvider;
+
+    private SimpleDateFormat dateFormat = new SimpleDateFormat(DateTimeType.DATE_PATTERN_WITH_TZ_AND_MS_GENERAL);
```

Comment:
Can this be a constant?

Reply:
Do you mean to make it final?

Reply:
Yes, and make it `static`. `SimpleDateFormat` is also not thread-safe. You need to synchronize access if it can be accessed by multiple threads concurrently.

Reply:
ok. I don't see any reason here to have multithreaded calls here, so I'd not add synchronized.